analyzers/src/SonarAnalyzer.CSharp/Rules/ForLoopCounterChanged.cs

```diff
@@ -83,10 +84,13 @@ private sealed class SideEffectExpression
 
                     foreach (var affectedExpression in AffectedExpressions(forNode.Statement))
                     {
-                        var symbol = c.SemanticModel.GetSymbolInfo(affectedExpression).Symbol;
+                        var symbol = TupleExpressionSyntaxWrapper.IsInstance(affectedExpression)
+                            ? TupleArgumentSymbolMatchingLoopCounter((TupleExpressionSyntaxWrapper)affectedExpression, loopCounters, c.SemanticModel)
```

Comment:
A more natural choice would be to operate with a list of symbols here (`var symbol=`). Maybe you can use `AllArguments` from here https://github.com/SonarSource/sonar-dotnet/compare/master...Martin/Decon_S1226 filter by name and convert it into symbols.

Reply:
I'm not 100% sure I understood properly, but I think I did something like that.